packages-content-model/roosterjs-content-model-dom/lib/modelApi/common/normalizeParagraph.ts

```diff
@@ -47,3 +49,12 @@ function removeEmptySegments(block: ContentModelParagraph) {
         }
     }
 }
+
+function removeEmptyLinks(paragraph: ContentModelParagraph) {
+    const segments = paragraph.segments;
+    const noMarkerSegments = segments.filter(x => x.segmentType != 'SelectionMarker');
+    const marker = segments.find(x => x.segmentType == 'SelectionMarker');
+    if (marker && marker.link && noMarkerSegments.every(x => !x.link)) {
+        delete marker.link;
```

Comment:
what if
```html
<a href="..">test1</a>
test2
[SELECTION MARKER HERE]
```
?

Reply:
You can do something like:

paragraph.segments.forEach((segment, i) => {
  if (segment.link && areSameFormat(segment.link, paragraph.segments[i-1]?.link) && areSameFormat(segment.ink, paragraph.segments[i+1]?.link) {
   delete segment.link;
}
}

Reply:
I think we only need to delete segment.link when the segment is selection marker. I think we might lose the link in other segment types if we check all segments.